saskatoon/harvest/api.py

```diff
@@ -221,3 +224,188 @@ def list(self, request, *args, **kwargs):
                                  'title': _("New Person")
                                  }
                          })
+
+
+class StatsViewset(LoginRequiredMixin, viewsets.ModelViewSet):
+
+    template_name = "app/stats.html"
+    queryset = Harvest.objects.filter(status="Succeeded").order_by("-id")
+    permission_classes = [IsCoreOrAdmin]
+    serializer_class = HarvestSerializer
+    filter_backends = (filters.DjangoFilterBackend,)
+    filterset_class = HarvestFilter
+    filterset_fields = (
+        "pick_leader",
+        "owner_fruit",
+        "property__neighborhood",
+        "property",
+        "status",
+        "season",
+    )
+    # ? season name that will be passed to template if no season is specified
+    season_name = "all"
+    # ? seasons that will be shown in the dropdown list
+    # TODO Use the original seasons ChoiceField in dropdown
+    seasons = list(map(lambda x: x[0], filterset_class.seasons))
+
+    def list(self, request, format="html", *args, **kwargs):
+        """returns statistics on harvests for all seasons or a specific season"""
+
+        season_parameter = self.kwargs.get("season")
+        if season_parameter is not None:
+            if season_parameter in self.seasons:
+                self.queryset.filter(start_date__year=season_parameter)
```

Comment:
what is this line doing? I don't think you are actually modyfing the queryset here?
Have you looked at the `get_queryset` method? 
But either way I am not sure I understand where you are re-using `self.queryset` to compute your stats